docs/calyx-py.md

```diff
@@ -1,17 +1,120 @@
+
+We use the `Builder` object to construct our Calyx program `prog`. We can define components with `prog.component`. Here's a defininition of a `main` component with a 32-bit input and output:
+
+```python
+{{#include ../calyx-py/test/builder_example.py:init}}
 ```
-python calyx-py/test/example.py
+
+Technically, we didn't need to assign `prog.component("main")` to a variable; the `main` component would have been added to `prog` regardless. However, it will often prove useful to store handles to components, registers, or other objects you'd like to use later.
+
+We then instantiate our cells: three 32-bit registers and one 32-bit adder.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:cells}}
+```
+
+As with adding components to a program, we don't need to assign `main.reg(...)` to a variable, but it'll be useful to be able to quickly refer to these cells.
+
+Next, we'll define our groups of assignments. The syntax for defining a group looks like `with {component}.group("group_name") as group_variable`, as we do below:
+
+```python
+{{#include ../calyx-py/test/builder_example.py:group_def}}
 ```
 
+Now, we'll initialize our registers. You can access cell ports using dot notation. Notably, port names that are also reserved keywords in Python such as `in` are followed by an underscore.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:assigns}}
+```
+
+As mentioned in the comments above, the Calyx builder will try to infer the bitwidth of constants. In case this doesn't work and you run into problems with this, you can provide the constant's size like so:
+
+```python
+{{#include ../calyx-py/test/builder_example.py:const}}
+```
+
+Calyx groups use a latency-insensitive go/done interface. Writing a group's done condition with the builder is pretty similar to doing so in Calyx, except that the `?` used for guarded assignments is now `@` (due to conflicting usage of `?` in Python).
+
+```python
+{{#include ../calyx-py/test/builder_example.py:done}}
+```
+
+In order to use the ports of cells in our `main` component within the code for our component, we'll expose the adder's output port by explicitly constructing it using the `calyx-py` AST.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:bare}}
+```
+
+Now, when we want to use the output port of our adder, we can do so easily:
+
+```python
+{{#include ../calyx-py/test/builder_example.py:bare_use}}
+```
+
+In order to add continuous assignments to your program, use the construct `with {component}.continuous:`.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:continuous}}
+```
+
+To access a component's ports while defining it, like we did above, we use the method `this()`.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:this}}
+```
+
+Lastly, we'll construct the control portion of this Calyx program. It's pretty simple; we're sequencing two groups. Sequences of groups are just Python lists:
```

Comment:
"sequencing two groups" confused me. Just say "running two groups in sequence"?

Reply:
https://github.com/cucapra/calyx/pull/1229#discussion_r1014495833 talks about lists for `seq` and sets for `par`. Can you check if that was implemented, and if so, document it?